db/migrate/20180501093846_create_events.rb

```diff
@@ -0,0 +1,39 @@
+# This migration does not use Rails ActiveRecord ORM DSL since
+# it doesn't provide data integrity (foreign key) for polymorphic models.
+#
+# The technique applied is called Exclusive Belongs To (AKA Exclusive Arc)
+#
+# Please read the following article for more details:
+# https://hashrocket.com/blog/posts/modeling-polymorphic-associations-in-a-relational-database
+#
+class CreateEvents < ActiveRecord::Migration
+  def up
+    execute <<-SQL
+      CREATE TABLE events (
+        id          serial PRIMARY KEY,
+        action      integer NOT NULL,
+        post_id     integer REFERENCES posts,
+        member_id   integer REFERENCES members,
+        transfer_id integer REFERENCES transfers,
+        created_at  timestamp without time zone,
+        updated_at  timestamp without time zone,
```

Comment:
I left I comment about this already but it got collapsed. What about having the regular create_table and then an alter table for these checks that can't be done with AR? I think this would make it a bit more clear that we specify things AR does not provide and perhaps a bit more readable.